core/src/main/java/org/mule/runtime/core/privileged/processor/MessageProcessors.java

```diff
@@ -307,35 +380,45 @@ public static BaseEventContext newChildContext(CoreEvent event, Optional<Compone
         })
         .transform(processor)
         .doOnNext(completeSuccessIfNeeded())
-        .mergeWith(Flux.<CoreEvent>create(errorSwitchSinkSinkRef))
+        .map(event -> right(MessagingException.class, event))
+        // This Either here is used to propagate errors. If the error is sent directly through the merged with Flux, it will be
+        // cancelled, ignoring the onErrorcontinue of the parent Flux.
+        .mergeWith(Flux.<Either<MessagingException, CoreEvent>>create(errorSwitchSinkSinkRef))
+        .map(result -> result.reduce(me -> {
+          throw propagateWrappingFatal(me);
+        }, response -> response))
         .distinct(event -> (BaseEventContext) event.getContext(), () -> seenContexts)
-        .map(result -> quickCopy((((BaseEventContext) result.getContext()).getParentContext().get()), result));
+        .map(result -> toParentContext(result));
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Replace this lambda with a method reference. ([squid:S1612](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS1612))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7683&issues=AWjyMA29otIXnmsUSXGh&open=AWjyMA29otIXnmsUSXGh)

[ISSUE_KEY]: AWjyMA29otIXnmsUSXGh